boot/spring_performance.go

```diff
@@ -0,0 +1,261 @@
+				return layer, fmt.Errorf("error copying jar\n%w", err)
+			}
+
+			jarPath = tempJarPath
+			os.RemoveAll(s.AppPath)
+		}
+
+		if err := s.springBootJarCDSLayoutExtract(jarPath); err != nil {
+			return layer, fmt.Errorf("error extracting Boot jar at %s\n%w", jarPath, err)
+		}
+		startClassValue, _ := s.Manifest.Get("Start-Class")
+
+		err := resetCreationTimeWithTouch(s)
+		if err != nil {
+			return libcnb.Layer{}, fmt.Errorf("error resetting file creation time for CDS\n%w", err)
+		}
+
+		trainingRunArgs = append(trainingRunArgs,
+			"-Dspring.context.exit=onRefresh",
+			"-XX:ArchiveClassesAtExit=application.jsa",
+			"-cp",
+		)
+		trainingRunArgs = append(trainingRunArgs, s.ClasspathString)
+		trainingRunArgs = append(trainingRunArgs, startClassValue)
+
+		var trainingRunEnvVariables []string
+
+		javaToolOptions := sherpa.GetEnvWithDefault("CDS_TRAINING_JAVA_TOOL_OPTIONS", sherpa.GetEnvWithDefault("JAVA_TOOL_OPTIONS", ""))
+		if javaToolOptions != "" {
+			s.Logger.Bodyf("Training run will use this value as JAVA_TOOL_OPTIONS: %s", javaToolOptions)
+			trainingRunEnvVariables = append(trainingRunEnvVariables, fmt.Sprintf("JAVA_TOOL_OPTIONS=%s", javaToolOptions))
+		}
+
+		// perform the training run, application.dsa, the cache file, will be created
+		if err := s.Executor.Execute(effect.Execution{
+			Command: "java",
+			Env:     trainingRunEnvVariables,
+			Args:    trainingRunArgs,
+			Dir:     s.AppPath,
+			Stdout:  s.Logger.InfoWriter(),
+			Stderr:  s.Logger.InfoWriter(),
+		}); err != nil {
+			return libcnb.Layer{}, fmt.Errorf("error running build\n%w", err)
+		}
+
+		return layer, nil
+	})
+
+	if err != nil {
+		return libcnb.Layer{}, fmt.Errorf("unable to contribute spring-cds layer\n%w", err)
+	}
+	return layer, nil
+}
+
+func resetCreationTimeWithTouch(s SpringPerformance) error {
+	// we set the creation date to the buildpack default 1980/01/01 date; the date the layer will appear to be created at
+	return s.Executor.Execute(effect.Execution{
+		Command: "find",
+		Env:     []string{"TZ=UTC"},
+		Args:    []string{"./", "-exec", "touch", "-t", "198001010000.01", "{}", ";"},
+		Dir:     s.AppPath,
+		Stdout:  s.Logger.InfoWriter(),
+		Stderr:  s.Logger.InfoWriter(),
+	})
+}
+
+func (s SpringPerformance) Name() string {
+	return s.LayerContributor.Name
+}
+
+func (s SpringPerformance) springBootJarCDSLayoutExtract(jarPath string) error {
+	s.Logger.Bodyf("Extracting Jar")
+	if err := s.Executor.Execute(effect.Execution{
+		Command: "java",
+		Args:    []string{"-Djarmode=tools", "-jar", jarPath, "extract", "--destination", s.AppPath},
+		Dir:     filepath.Dir(jarPath),
+		Stdout:  s.Logger.InfoWriter(),
+		Stderr:  s.Logger.InfoWriter(),
+	}); err != nil {
+		return fmt.Errorf("error extracting Jar with jarmode\n%w", err)
+	}
+	return nil
+}
+
+// CreateJar heavily inspired by: https://gosamples.dev/zip-file/
+func CreateJar(source, target string) error {
```

Comment:
If we have to keep this, I'd like to see it move into libpak under the crush module. It's going to keep things consistent with respect to handling archives. There's a CreateTar method in there, so seems reasonable to add a CreateZip or CreateJar method.

Reply:
done, using libpak 1.7.0